src/compiler/core.ts

```diff
@@ -837,6 +837,46 @@ export function sortAndDeduplicate<T>(array: readonly T[], comparer?: Comparer<T
     return deduplicateSorted(sort(array, comparer), equalityComparer || comparer || compareStringsCaseSensitive as any as Comparer<T>);
 }
 
+/**
+ * Merges two sorted arrays so that the resulting array is sorted and has no duplicates.
+ * @internal
+ */
+export function mergeAndDeduplicateSorted<T>(array1: readonly T[], array2: readonly T[], comparer: Comparer<T>): SortedReadonlyArray<T> {
+    if (array1.length === 0) return array2 as any as SortedReadonlyArray<T>;
+    if (array2.length === 0) return array1 as any as SortedReadonlyArray<T>;
+
+    const finalArray: T[] = [];
+    let i = 0, j = 0;
+    while (i < array1.length && j < array2.length) {
+        switch (comparer(array1[i], array2[j])) {
+            case Comparison.LessThan:

```

Comment:
This isn't something new, but I am somewhat nervous about switching on these values exactly; our Comparison type is unsound and we have cases where we actually return values that are not these three, but instead just positive/negative values. See also #52897.

Maybe it doesn't happen with what is passed to this new function (and the assert helps), of course.

Reply:
I actually encountered that bug already while working on this function (fixed by using a better `comparer`), so I put the default option there to prevent any loose ends from happening with unsound Comparisons.